### ISSUE TITLE
Layout/EndAlignment for multi-line blocks

### ISSUE THREAD

**Issue by pirj** [rubocop/rubocop#7752]
`EnforcedStyleAlignWith: start_of_line` doesn't seem to be a valid option for multi-line statements.

## Steps to reproduce the problem
 
```ruby
@expected_number = case count
                   when :once then 1
                   else Integer(count)
                   end
```

[Origin](https://github.com/rspec/rspec-rails/blob/bd54ec4f9c63d9e36dbb3196a59b0908561e287d/lib/rspec/rails/matchers/active_job.rb#L167).

The cop suggests to align `end` with `@`, and that's an extremely odd suggestion both style-wise:
```ruby
@expected_number = case count
                   when :once then 1
                   else Integer(count)
end
```

and correctness-wise:
```ruby
      path = if uri.path == ''
               '/'
             else
               uri.path
      end
```
since Ruby 2.7 started complaining about such a style:
```
gems/cookiejar-0.3.3/lib/cookiejar/cookie.rb:183: warning: mismatched indentations at 'end' with 'else' at 181
```

### Side Observation

`variable` in `EnforcedStyleAlignWith: variable` is confusing, since the cop operates on any `send`s, not only on `lvasgn`:
```ruby
puts case var
     when 1
       2
     else
       3
     end
```
Shouldn't it be `receiver`?

For obvious reasons, this does not apply to `if var`, however it should for `case`/`begin`/other Ruby multi-line constructs.

Related to: #6774, #7337

## Expected behavior

- [ ] `start_of_line` and ~`variable`~ `receiver` are not considered/allowed options for multi-line statements.

## Actual behavior

`variable`/`start_of_line` raise an offence for multi-line statements for `start_of_line` and `variable` enforced styles.

## RuboCop version

```
$ [bundle exec] rubocop -V
0.78.0 (using Parser 2.6.5.0, running on ruby 2.7.0 x86_64-darwin18)
```

**Comment by robotdana** [comment]
What is wrong with this format? 
```ruby
value = case whatever
when 1
   2
else
  3
end
```

**Comment by pirj** [comment]
Is there a comparable cop for `if`/`begin` @robotdana ?

**Comment by robotdana** [comment]
there must be because i use the same pattern for if, begin, etc.

**Comment by pirj** [comment]
Stale, but bug.

**Comment by mjobin-mdsol** [comment]
How about enforcing 

```ruby
value = 
  case whatever
  when 1
     2
  else
    3
  end
```

isn't that also popular choice?

**Comment by jonas054** [comment]
Like @robotdana initially remarked, the key is to combine the configuration with `Layout/CaseIndentation: EnforcedStyle: end`.

With
```yaml
Layout/EndAlignment:
  EnforcedStyleAlignWith: start_of_line

Layout/CaseIndentation:
  EnforcedStyle: end
```

then `rubocop -A` corrects code into
```ruby
@expected_number = case count
when :once
  1
else
  Integer(count)
end

@expected_number = if count == :once
  1
else
  Integer(count)
end

@expected_number = begin
  result = count == :once ? 1 : Integer(count)
  result
end
```

which I think is fine, and as indented. @pirj I do agree with your side comment that there is something fishy about how `Layout/EndAlignment: EnforcedStyleAlignWith: variable` works, but I think that should not be solved here, because it's a separate discussion.

I don't see any bug, so I suggest we close this issue.
